librepo/yum.c

```diff
@@ -152,9 +162,29 @@ lr_yum_repomd_record_enabled(LrHandle *handle, const char *type)
                 return TRUE;
             x++;
         }
+        // Substitution check
+        if (handle->yumslist) {
+            LrUrlVars* elem = handle->yumslist;
+            while (TRUE) {
+                LrVar* subs = elem->data;
+                if (!strcmp(subs->var, type)) {
+                    char *orig = subs->val;
+                    int y = 0;
+                    while (handle->yumdlist[y]) {
+                        if (!strcmp(orig, handle->yumdlist[y]) &&
```

Comment:
Here you can also save lines 172, 173, 177 by:

    for (guint i = 0; handle->yumdlist[i]; i++) {